### PR TITLE
device: Remove z_device_is_ready

### PR DESCRIPTION
**Pull request by ceolin**
This duplicates the functionality of device_is_ready.

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

File: kernel/device.c [hunk 1]
```diff
@@ -40,13 +40,13 @@ const struct device *z_impl_device_get_binding(const char *name)
 	 * performed. Reserve string comparisons for a fallback.
 	 */
 	STRUCT_SECTION_FOREACH(device, dev) {
-		if (z_device_is_ready(dev) && (dev->name == name)) {
+		if (z_impl_device_is_ready(dev) && (dev->name == name)) {
```
Comment (nashif) [inline kernel/device.c:43]: why are you calling the implementation directly here and not using `device_is_ready` instead?

**Review by nashif — CHANGES_REQUESTED** [review]

File: kernel/device.c (hunk 1 above)
Comment (ceolin) [reply kernel/device.c:43]: yep

**Review by cfriedt — APPROVED** [review]

File: kernel/device.c (hunk 1 above)
Comment (nashif) [reply kernel/device.c:43]: do we agree that calling implementation (z_impl) in general should not be the norm and that any direct usage of the implementation function need some justification? If yes, this justifcation should be in the commit message, i.e. what @dcpleung  wrote above.

Comment (ceolin) [reply kernel/device.c:43]: @nashif I have update the commit msg. Please take a look again.

**Review by nashif — APPROVED** [review]

**Pull request merged**